src/action/action.ts

```diff
@@ -0,0 +1,440 @@
+    const payload: UnencodedActionPayload<PayloadType.EXECUTE_ACTION> = {
+      dbid: dbid,
+      action: actionName,
+      arguments: preparedActions,
+    };
+
+    // build the transaction
+    return await Payload.createTx(this.kwil, {
+      payloadType: PayloadType.EXECUTE_ACTION,
+      payload: payload,
+      signer: signer,
+      signatureType: signatureType,
+      description: this.description,
+      challenge: this.challenge,
+      signature: this.signature,
+      chainId: chainId,
+      identifier: identifier as Uint8Array, // Fix this. Seems hacky...
+      nonce: this.nonce,
+    })
+      .buildTx()
+      .finally(() => (this.actionInputs = cached));
+  }
+
+  /**
+   * Builds the message structure for view actions. This can be provided to the `kwil.call()` api.
+   */
+  async buildMsg(): Promise<Message> {
+    this.assertNotBuilding();
+
+    // cache the action
+    const cached = this.actionInputs;
+
+    // set the actions to a special value to indicate that the message is being built
+    this.actionInputs = TXN_BUILD_IN_PROGRESS;
+
+    // retrieve the schema and run validations
+    const { dbid, actionName, preparedActions, modifiers } = await this.checkSchema(
+      cached as ActionInput[]
+    );
+
+    // throw a runtime error if more than one set of inputs is trying to be executed. Call does not allow bulk execution.
+    if (preparedActions && preparedActions.length > 1) {
+      throw new Error(
+        'Cannot pass more than one input to the call endpoint. Please pass only one input and try again.'
+      );
+    }
+
+    // throw runtime error if action is not a view action. transactions require a different payload structure than view actions.
+    if (modifiers && modifiers.includes('VIEW') === false) {
+      throw new Error(`Action ${actionName} is not a view only action. Please use kwil.execute().`);
+    }
+
+    // resolve if a signer is or is not specified. Only actions with `mustsign` require a signer.
+    const signer = this.signer ? this.signer : null;
+
+    // construct payload. If there are no prepared actions, then the payload is an empty array.
+    const payload: UnencodedActionPayload<PayloadType.CALL_ACTION> = {
+      dbid: dbid,
+      action: actionName,
+      // if there are prepared actions, then the first element in the array is the action inputs.
+      arguments: preparedActions.length > 0 ? preparedActions[0] : [],
+      // if there are nilArgs, then the first element in the array is the nilArgs.
+    };
+
+    let msg = await Payload.createTx(this.kwil, {
+      payload: payload,
+      challenge: this.challenge,
+      signature: this.signature,
+    });
+
+    // if a signer is specified, add the signer, signature type, identifier, and description to the message
+    if (signer) {
+      const identifier = await Promisy.resolveOrReject(this.identifier as Nillable<string>);
+      const signatureType = await Promisy.resolveOrReject(this.signatureType);
```

Comment:
Same here - no need to `Promisy.resolveOrReject`. This.signatureType does not need to be resolved.

I can explain `Promisy` was originally for offline, but the main point is: Bcz most args to it are now internal (and the args you are passing are resolved), they aren't needed anymore.